Ccp 1959 - Welcome to Swell Widget

dqdquint opened the pull request:
Shop No: 31912 was used to test and there is a test widget on the home page.

Added info on configurations in Demo file and Readme.

Commit:
Added configurable widget: TODO: History Button Modal

Commit:
Merged updated dev

Commit:
Using showPopupByTye from Swell SDK

Commit:
Corrected Readme file

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedInView/index.jsx:
```diff
@@ -0,0 +1,68 @@
+  return (
+    <div className={styles.container}>
+      <I18n.Text className={styles.heading(headerTextColor)} string="swell.welcome.greeting" params={{ firstName }} />
+      <I18n.Text className={styles.heading(headerTextColor)} string="swell.welcome.points" params={{ points }} />
```
we should combine both into one translation string to be more flexible

dqdquint replied:
I'm not sure they should be one translation. Not sure how to style the locale string since new line character is ignored when I tested implementing this.

Carsten89 commented on frontend/widgets/SwellWelcome/index.jsx:
```diff
@@ -0,0 +1,75 @@
+  settings,
+  points,
+}) => {
+  if (isFetching) {
```
i removed the isFetching state in CCP-1954. please remove it here too

dqdquint replied:
Removed

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedOutView/style.js:
```diff
@@ -0,0 +1,70 @@
+ */
+const container = (color, backgroundImage) => css({
+  color,
+  backgroundImage,
```
```suggestion
  background,
```
in settings:
"loggedOutBackgroundImage" --> loggedOutBackground

with this change it will be possible to use simple bg color too

Carsten89 replied:
we should also add background-size cover in the default value for "loggedOutBackground"

dqdquint replied:
Updated. Thanks for the suggestion.

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedInView/index.jsx:
```diff
@@ -0,0 +1,68 @@
+      <I18n.Text className={styles.heading(headerTextColor)} string="swell.welcome.points" params={{ points }} />
+      <div className={styles.buttonContainer}>
+        <Link
+          href={CHECKOUT_PATH}
```
on drinkhint they link to the cart page. not checkout

dqdquint replied:
Updated to use cart path. thanks!

Carsten89 commented on frontend/widgets/SwellWelcome/index.jsx:
```diff
@@ -0,0 +1,75 @@
+/**
+ * @returns {JSX}
+ */
+const SwellWelcome = ({
```
we should add some top padding
![image](https://user-images.githubusercontent.com/5811286/74317173-2da00780-4d7b-11ea-826b-beaac109e3e6.png)

dqdquint replied:
Added a margin config to settings

Carsten89 committed:
small fixes

Commit:
Changes for reducer and subscription, merged new dev branch

Commit:
Added white space

Commit:
Addressed PR feedback

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedOutView/style.js:
```diff
@@ -0,0 +1,72 @@
+const joinButton = css({
+  fontSize: '18px',
+  marginBottom: '10px',
+  backgroundColor: '#002554',
```
this should be a config too. didn't see this in prev review, sorry.

aaron-martin replied:
made configurable

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedOutView/style.js:
```diff
@@ -0,0 +1,72 @@
+  fontSize: '18px',
+  marginBottom: '10px',
+  backgroundColor: '#002554',
+  color: '#fff',
```
this should be a config too. didn't see this in prev review, sorry.

aaron-martin replied:
made configurable

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedOutView/style.js:
```diff
@@ -0,0 +1,72 @@
+  maxWidth: '175px',
+  padding: '10px',
+  borderRadius: '5px',
+  border: '1px solid #002554',
```
this should be a config too. didn't see this in prev review, sorry.

aaron-martin replied:
made configurable

Carsten89 commented on frontend/widgets/SwellWelcome/components/LoggedInView/index.jsx:
```diff
@@ -0,0 +1,71 @@
+  headerTextColor: PropTypes.string.isRequired,
+  historyButtonText: PropTypes.string.isRequired,
+  redeemButtonText: PropTypes.string.isRequired,
+  widgetMargin: PropTypes.string.isRequired,
```
a lot of these props should be optional for me

Commit:
CCP-1959: responded to feedback

Commit:
CCP-1959: merge dev into this branch

# Conflicts:
#	README.md
#	extension-config.json
#	frontend/constants.js
#	frontend/locale/en-US.json
#	frontend/reducers/index.js
#	frontend/selectors/index.js
